### PR TITLE
Add Manage Whitelist dialog

### PR DESCRIPTION
**Pull request by ArmandoGraterol**
Added manage whitelist dialog and updated some of the related components like `CSVUploader`.

FireShot Capture 796 - Colony CDapp - localhost
FireShot Capture 797 - Colony CDapp - localhost
FireShot Capture 798 - Colony CDapp - localhost

Contributes to #58

### CODE REVIEW

File: src/components/common/Dialogs/ManageWhitelistDialog/ManageWhitelistDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+interface CustomWizardDialogProps {
+  colony: Colony;
+  userAddress?: string;
+}
+
+type Props = DialogProps &
+  Partial<WizardDialogType<object>> &
+  CustomWizardDialogProps;
+
+const displayName = 'common.ColonyHome.ManageWhitelistDialog';
```
Comment (rdig) [inline src/components/common/Dialogs/ManageWhitelistDialog/ManageWhitelistDialog.tsx:44]: This should be under `Dialogs` not `ColonyHome` here as well

File: src/components/common/Dialogs/ManageWhitelistDialog/NoWhitelistedAddressesState/NoWhitelistedAddressesState.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { FormattedMessage, defineMessages } from 'react-intl';
+
+import styles from './NoWhitelistedAddressesState.css';
+
+const displayName =
+  'common.ColonyHome.ManageWhitelistDialog.NoWhitelistedAddressesState';
```
Comment (rdig) [inline src/components/common/Dialogs/ManageWhitelistDialog/NoWhitelistedAddressesState/NoWhitelistedAddressesState.tsx:7]: This should be in `Dialogs` too

File: src/components/common/Dialogs/ManageWhitelistDialog/WhitelistedAddresses/WhitelistedAddresses.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+import styles from './WhitelistedAddresses.css';
+
+interface Props {
+  colony: Colony;
+  whitelistedUsers: User[];
+}
+
+const displayName =
+  'common.ColonyHome.ManageWhitelistDialog.WhitelistedAddresses';
```
Comment (rdig) [inline src/components/common/Dialogs/ManageWhitelistDialog/WhitelistedAddresses/WhitelistedAddresses.tsx:19]: Same here about being in `Dialogs`

**Review by rdig — APPROVED** [review]
Besides component names, this is mostly good, however I'm still unsure how are we going to handle Whitelist with the database backend.

For now feel free to leave it in.

**Review by Julianahlk — CHANGES_REQUESTED** [review]
I am seeing a few issues after uploading a .csv

- The confirm button is still disabled, I would either expect it to be enabled or display an error message explaining what is wrong
- The back button is displayed as "button.back"
- If you switch tabs to Address Book then return back to the Add Address tab, the state in the UI of the uploaded csv is not saved.

Screen Shot 2023-02-16 at 10 02 18 PM


- Uploading the template.csv that does not contain any whitelisted addresses triggers an uncaught exception

Screen Shot 2023-02-16 at 10 09 44 PM

**Review by jakubcolony — CHANGES_REQUESTED** [review]
@ArmandoGraterol I'm seeing similar issues to Juliana with the back button, but more importantly, when I tried to drop the unmodified template, I got stuck in render loop:
image

When I tried to replicate it however, all I was seeing is the same error that Juliana already reported.

Does it make sense for the textarea to be shown when there are no addresses in the book?
image

I'll re-review once the above is addressed

File: src/components/common/Dialogs/ManageWhitelistDialog/ManageWhitelistActiveToggle.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+
+import { Toggle } from '~shared/Fields';
+import { Heading4 } from '~shared/Heading';
+
+import styles from './ManageWhitelistDialogForm.css';
+
+const displayName = `common.ManageWhitelistDialog.ManageWhitelistDialogForm.ManageWhitelistActiveToggle`;
```
[the remark this answers is not in the document]
Comment (ArmandoGraterol) [reply src/components/common/Dialogs/ManageWhitelistDialog/ManageWhitelistActiveToggle.tsx:9]: If by path you mean the folder path, that's correct, but this component is only used inside `ManageWhitelistDialogForm`. It's like `NetworkFee` in that sense, it's not used on `CreatePaymentDialog` directly but in the form of the dialog.

File: src/components/common/Dialogs/ManageWhitelistDialog/ManageWhitelistActiveToggle.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    defaultMessage: 'Only show address book contacts as recipients',
+  },
+  tooltipText: {
+    id: `${displayName}.tooltipText`,
+    defaultMessage: `Address book is active by default once at least one address is added to the list. You can turn this feature “Off” to deactivate. Use with caution.`,
+  },
+  warningText: {
+    id: `${displayName}.warningText`,
+    defaultMessage: `<span>Warning.</span>
+      You are deactivating your colony’s address book.
```
[the remark this answers is not in the document]
Comment (ArmandoGraterol) [reply src/components/common/Dialogs/ManageWhitelistDialog/ManageWhitelistActiveToggle.tsx:30]: Sounds good to me x2

**Comment by ArmandoGraterol** [comment]
@Julianahlk @jakubcolony Added some changes to could solve the issues that you guys are having, please let me know if it works! Working on this CSV uploader is a bit of a hassle.
 
> * If you switch tabs to Address Book then return back to the Add Address tab, the state in the UI of the uploaded csv is not saved.

That's intentional per line 114 in `ManageWhitelistDialogForm`. Do you think that it would be best that we don't reset the form when we change tabs? I don't have a strong opinion on it :)

**Review by jakubcolony — APPROVED** [review]
LGTM, all seems to work 👍 Nice job!

**Review by Julianahlk — APPROVED** [review]
Working much better and approving for now.

Regarding saving the state when switching between the add address and address book tabs, in an ideal world it would be nice if the state saved because if you upload a csv then switch tabs, you need to upload it again.

**Pull request merged**